gradle-plugin/plugin/src/main/resources/kotlin/tools/TypesAdapters.kt.mustache

```diff
@@ -60,7 +60,7 @@ internal class LocalDateAdapter : XNullableJsonAdapter<LocalDate>() {
     override fun fromNonNullString(nextString: String): LocalDate = LocalDate.parse(nextString, formatter)
 
     override fun toJson(writer: JsonWriter, value: LocalDate?) {
-        value?.let { writer.value(it.format(formatter)) }
+        value?.let { writer.value(it.format(formatter)) } ?: writer.nullValue()
```

Comment:
I would rather suggest to have concrete implementations dealing only with the "not-nullable" case and having `XNullableJsonAdapter` taking care of the `null` handling.

Something like
```kotlin
internal abstract class XNullableJsonAdapter<T> : JsonAdapter<T>() { 
    abstract fun fromNonNullString(nextString: String): T
    abstract fun toJsonNonNull(writer: JsonWriter, value: T)
    ...
}
```
This allows to focus the attention of nullable in the class related to nullable and to the `T` type encoding in the related adapter class